### PR TITLE
221 status mapping

### PR DESCRIPTION
**Pull request by SPTKL**
- pending on dcp_application
- all other
- #221 
- #217

### CODE REVIEW

**Review by mgraber — COMMENTED** [review]
This makes sense to me. In the existing workflow, we have to apply this same status logic (as well as the boolean flag logic and phasing logic) after both rounds of review. This is because we have to set these attributes for records that are added/modified through the corrections process. So while this makes the data-loading step simpler, we will still have to duplicate that portion of the code, right? 

See these lines: https://github.com/NYCPlanning/db-knownprojects/blob/dev/knownprojects_build/05_corrections.sh#L3-L11

**Comment by SPTKL** [comment]
records will be added upstream before all those fields are assigned, so we don't have to repeat code

File: knownprojects_build/_sql/combine.sql [hunk 1]
```diff
@@ -71,24 +124,32 @@ _dcp_planneradded as (
...
+        flag_gq(a::text) as gq,
+        flag_senior_housing(a::text) as senior_housing,
+        flag_assisted_living(a::text) as assisted_living
+    FROM dcp_planneradded a
 ),
 _dcp_n_study_future as (
     SELECT
         'Future Neighborhood Studies' as source, 
         a.uid as record_id,
         array_agg(a.uid) as record_id_input,
```
Comment (mgraber) [inline knownprojects_build/_sql/combine.sql:138]: Does it make sense to have the array_agg here if a.uid is in the group by statement?

File: knownprojects_build/_sql/combine.sql [hunk 2]
```diff
@@ -160,7 +233,7 @@ _hpd_pc as (
...
```
Comment (mgraber) [inline knownprojects_build/_sql/combine.sql:234]: Same comment as above. a.uid is in the group by statement.

File: knownprojects_build/_sql/combine.sql [hunk 3]
```diff
@@ -44,15 +92,19 @@ _edc_projects as (
...
```
Comment (mgraber) [inline knownprojects_build/_sql/combine.sql:93]: Same here.

**Review by mgraber — COMMENTED** [review]
In `combined`, there are cases where record_id_input contains a single id. Sometimes this is the same ID as the `record_id`, which I think is happening in the cases where uid is in the group by statement. Look at edc_projects for example. Should we be including the single record for all sources that don't get rolled up? Some are NULL instead.

There are also cases where record_id_input contains a single value, and this is different than record_id, since we're taking the hash of the array_agg. An example is hpd_rfp. Should we be doing this for all sources?

**Comment by mgraber** [comment]
Perhaps we can use a similar convention to keep track of clusters:
```sql
SELECT 
	array_agg(a.record_id) as project_id_input
FROM(
	SELECT record_id, 
		coalesce(ST_ClusterDBSCAN(geom, 0, 1) OVER(), 99999) AS cluster_id
	FROM  combined
) a
GROUP BY cluster_id;
```

**Comment by SPTKL** [comment]
@mgraber  I don't know how to address this yet, I can't find a function that creates a 1 item array, it could also just be something like array_append([]text, uid) for these cases. anyways this PR is for status mapping, we can address this issue later

**Pull request merged**